tests/rest/client/test_rooms.py

```diff
@@ -968,6 +995,63 @@ async def user_may_join_room(
         return_value = False
         self.helper.join(self.room3, self.user2, expect_code=403, tok=self.tok2)
 
+    def test_spam_checker_may_join_room(self) -> None:
+        """Tests that the user_may_join_room spam checker callback is correctly called
+        and blocks room joins when needed.
+        """
+
+        # Register a dummy callback. Make it allow all room joins for now.
+        return_value: Union[Literal["NOT_SPAM"], Codes] = "NOT_SPAM"
```

Comment:
`"NOT_SPAM"` should be replaced with a reference to the constant modules will actually use.